RestApi/Sepes.RestApi/Controllers/AuthController.cs

```diff
@@ -1,121 +1,44 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
 using System.IdentityModel.Tokens.Jwt;
-using System.Text;
-using System.Text.Json;
-using System.Web;
-using System.Security.Claims;
-using System.Threading.Tasks;
 using Microsoft.AspNetCore.Mvc;
-using Microsoft.AspNetCore.Authentication;
-using Microsoft.AspNetCore.Authentication.JwtBearer;
 using Microsoft.AspNetCore.Authorization;
-using Microsoft.ApplicationInsights;
-using Microsoft.ApplicationInsights.DataContracts;
-using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Options;
 using Microsoft.IdentityModel.Tokens;
 using Sepes.RestApi.Model;
+using Sepes.RestApi.Services;
+using System.Threading.Tasks;
 
 namespace Sepes.RestApi.Controller
 {
     [Route("api/[controller]")]
     [ApiController]
     public class AuthController : ControllerBase
     {
-
-        //Make values from appsetting availible to this code
-        private AppSettings _appSetting;
-        public AuthController(IOptions<AppSettings> appsettings)
+        public readonly IAuthService _auth;
+        public AuthController(IAuthService authService)
         {
-            _appSetting = appsettings.Value;
+            _auth = authService;
         }
-         //POST api/auth/token
+
+        //POST api/auth/token
         [AllowAnonymous]
         [HttpPost("token")]
-        public IActionResult Token([FromBody] AzTokenClass AzToken) 
+        public async Task<string> Token([FromBody] AzTokenClass AzToken)
         {
-            IActionResult response = Unauthorized(); //Default response if bellow fails
-            var IsAuthentic = AuthenticateToken(AzToken);
-            
-            if (IsAuthentic){
-                var SepesToken = GenerateJSONWebToken(AzToken, null);
-                //response = Ok(new { Token = SepesToken}); //Alternative method to return wrapped in token. In case we need more attributes later.
-                response = Ok(SepesToken);
-            }
-            else{
-                //Issue 41: look into error handling for identifiable conditions. Ex. if azure verification timed out.
-                response = Unauthorized("Invalid Azure token.");
-            }
-
-
-            
-            //Check if creation of token is automatically logged, if not add custom logging that reports what userid got what token.
-            
-            return response;
+            ActionResult response = Unauthorized(); //Default response if bellow fail
+            var SepesToken = await _auth.GenerateJSONWebToken(AzToken.idToken, null);
```

Comment:
When you call the same function with to parameters where one is null. And another time with the arguments reversed. That is hint that is actually 2 functions.